<!-- (x -!dopredu +!dozadu) (y -levo +vpravo) (z -dolu +nahoru) -->
<robot name="robik">

	<link name="base_link">
		<visual>
			<origin xyz="0 0 0.06" rpy="1.57 0 -1.57" />
			<geometry>
				<mesh filename="package://robik/model/robot_model_v2.dae"
					scale="1.0 1.0 1.0" />
			</geometry>
			<material name="clear">
				<color rgba="1 1 1 1" />
			</material>
		</visual>
	</link>

	<joint name="laser_joint" type="fixed">
		<parent link="base_link" />
		<child link="laser_link" />
		<origin xyz="-0.25 0 0.08" rpy="0 0 3.1415" />
	</joint>

	<link name="laser_link" />


	<link name="yaw_link">
		<visual>
			<geometry>
				<box size="0.045 0.04 0.06" />
			</geometry>
			<material name="Cyan">
				<color rgba="0 1.0 1.0 1.0" />
			</material>
		</visual>
	</link>

	<joint name="yaw_joint" type="revolute">
		<parent link="base_link" />
		<child link="yaw_link" />
		<origin xyz="0.02 -0.08 0.11" rpy="0 0 0" />
		<axis xyz="0 0 -1" />
		<limit lower="-1.8675023" upper="1.099557429" effort="1000" velocity="1" />
	</joint>


	<link name="shoulder_link">
		<visual>
			<origin xyz="0 0.0675 0" rpy="0 0 0"/>
			<geometry>
				<box size="0.04 0.135 0.02" />
			</geometry>
			<material name="Cyan">
				<color rgba="0 1.0 1.0 1.0" />
			</material>
		</visual>
	</link>

	<joint name="shoulder_joint" type="revolute">
		<parent link="yaw_link" />
		<child link="shoulder_link" />
		<origin xyz="0.0 0 0.03" rpy="0 0 -1.570796327" />
		<axis xyz="1 0 0" />
		<limit lower="0" upper="2.094395102" effort="1000" velocity="1" />
	</joint>


	<link name="elbow_link">
		<visual>
			<origin xyz="0 0.0325 0" rpy="0 0 0"/>
			<geometry>
				<box size="0.04 0.065 0.02" />
			</geometry>
			<material name="Cyan">
				<color rgba="0 1.0 1.0 1.0" />
			</material>
		</visual>
	</link>

	<joint name="elbow_joint" type="revolute">
		<parent link="shoulder_link" />
		<child link="elbow_link" />
		<origin xyz="0 0.135 0" rpy="0 0 0" />
		<axis xyz="1 0 0" />
		<limit lower="-1.919862177" upper="0.6981317008" effort="1000" velocity="1" />
	</joint>


	<link name="roll_link">
		<visual>
			<geometry>
				<box size="0.03 0.003 0.04" />
			</geometry>
			<material name="Cyan">
				<color rgba="0 1.0 1.0 1.0" />
			</material>
		</visual>
	</link>

	<joint name="roll_joint" type="revolute">
		<parent link="elbow_link" />
		<child link="roll_link" />
		<origin xyz="0 0.065 0" rpy="0 0 0" />
		<axis xyz="0 1 0" />
		<limit lower="-1.570796327" upper="1.570796327" effort="1000" velocity="1" />
	</joint>


	<link name="clamp_link">
		<visual>
			<origin xyz="0 0.025 0" rpy="0 0 0"/>
			<geometry>
				<box size="0.02 0.05 0.01" />
			</geometry>
			<material name="Cyan">
				<color rgba="0 1.0 1.0 1.0" />
			</material>
		</visual>
	</link>

	<joint name="clamp_joint" type="revolute">
		<parent link="roll_link" />
		<child link="clamp_link" />
		<origin xyz="0 0.001 -0.005" rpy="0 0 0" />
		<axis xyz="-1 0 0" />
		<limit lower="0" upper="0.6981317008" effort="1000" velocity="1" />
		<!--<mimic joint="clamp_joint" multiplier="-1" offset="0"/>-->
	</joint>

	<link name="left_wheel" />
	<link name="right_wheel" />

	<joint name="wheel_left_joint" type="continuous">
		<parent link="base_link"/>
		<child link="left_wheel"/>
	</joint>

	<joint name="wheel_right_joint" type="continuous">
		<parent link="base_link"/>
		<child link="right_wheel"/>
	</joint>

</robot>
